projects/plugins/jetpack/extensions/blocks/donations/deprecated/v2/index.js

```diff
@@ -0,0 +1,259 @@
+				amounts: [ 5, 15, 100 ],
+				heading: __( 'Make a one-time donation', 'jetpack' ),
+				extraText: __( 'Your contribution is appreciated.', 'jetpack' ),
+				buttonText: __( 'Donate', 'jetpack' ),
+			},
+		},
+		monthlyDonation: {
+			type: 'object',
+			default: {
+				show: true,
+				planId: null,
+				amounts: [ 5, 15, 100 ],
+				heading: __( 'Make a monthly donation', 'jetpack' ),
+				extraText: __( 'Your contribution is appreciated.', 'jetpack' ),
+				buttonText: __( 'Donate monthly', 'jetpack' ),
+			},
+		},
+		annualDonation: {
+			type: 'object',
+			default: {
+				show: true,
+				planId: null,
+				amounts: [ 5, 15, 100 ],
+				heading: __( 'Make a yearly donation', 'jetpack' ),
+				extraText: __( 'Your contribution is appreciated.', 'jetpack' ),
+				buttonText: __( 'Donate yearly', 'jetpack' ),
+			},
+		},
+		showCustomAmount: {
+			type: 'boolean',
+			default: true,
+		},
+		chooseAmountText: {
+			type: 'string',
+			default: __( 'Choose an amount', 'jetpack' ),
+		},
+		customAmountText: {
+			type: 'string',
+			default: __( 'Or enter a custom amount', 'jetpack' ),
+		},
+		fallbackLinkUrl: {
+			type: 'string',
+			default: '',
+		},
+	},
+	supports: {
+		html: false,
+	},
+	save: ( { attributes } ) => {
+		const { fallbackLinkUrl, oneTimeDonation, monthlyDonation, annualDonation } = attributes;
+
+		if (
+			! oneTimeDonation ||
+			! oneTimeDonation.show ||
+			! oneTimeDonation.planId ||
+			oneTimeDonation.planId === -1
+		) {
+			return null;
+		}
+
+		return (
+			<div>
+				<RichText.Content tagName="h4" value={ oneTimeDonation.heading } />
+				<RichText.Content tagName="p" value={ oneTimeDonation.extraText } />
+				<RichText.Content
+					tagName="a"
+					className="jetpack-donations-fallback-link"
+					href={ fallbackLinkUrl }
+					rel="noopener noreferrer noamphtml"
+					target="_blank"
+					value={ oneTimeDonation.buttonText }
+				/>
+				{ monthlyDonation.show && (
+					<>
+						<hr className="donations__separator" />
+						<RichText.Content tagName="h4" value={ monthlyDonation.heading } />
+						<RichText.Content tagName="p" value={ monthlyDonation.extraText } />
+						<RichText.Content
+							tagName="a"
+							className="jetpack-donations-fallback-link"
+							href={ fallbackLinkUrl }
+							rel="noopener noreferrer noamphtml"
+							target="_blank"
+							value={ monthlyDonation.buttonText }
+						/>
+					</>
+				) }
+				{ annualDonation.show && (
+					<>
+						<hr className="donations__separator" />
+						<RichText.Content tagName="h4" value={ annualDonation.heading } />
+						<RichText.Content tagName="p" value={ annualDonation.extraText } />
+						<RichText.Content
+							tagName="a"
+							className="jetpack-donations-fallback-link"
+							href={ fallbackLinkUrl }
+							rel="noopener noreferrer noamphtml"
+							target="_blank"
+							value={ annualDonation.buttonText }
+						/>
+					</>
+				) }
+			</div>
+		);
+	},
+	migrate: attributes => {
```

Comment:
I've found a migration issue :( I _don't think_ this was present when I first reviewed this PR.

The amounts are changing when opening up the editor with content created on the old version.

Previously I had $5, $15, $100, now in the editor I have $0, $1, $2. The placeholder for custom amount remains unchanged at $50

Repro steps:

 1. While on the master branch create a donations block
 2. Switch branches, rebuild blocks etc
 3. View the post on the front-end see the amounts haven't changed
 4. Open the post in the editor, see that the amounts have changed

Reply:
It's fixed in USD but not in other currencies.

 1. Create a new post on the master branch
 2. Add a donations block
 3. Set it to GBP, the values will be £3, £9 and £60, custom £30
 3. Save post
 4. Switch branch to this one and rebuild blocks
 5. Reload the editor
 6. See the values are now £1.80, £5.40 and £36.00. Custom £30

Doing the same thing in USD works fine.

Reply:
This should be working as expected now.